ci/lint/06_script.sh

```diff
@@ -6,13 +6,12 @@
 
 export LC_ALL=C
 
+GIT_HEAD=$(git rev-parse HEAD)
 if [ -n "$CIRRUS_PR" ]; then
-  # CIRRUS_PR will be present in a Cirrus environment. For builds triggered
-  # by a pull request this is the name of the branch targeted by the pull request.
-  # https://cirrus-ci.org/guide/writing-tasks/#environment-variables
-  COMMIT_RANGE="$CIRRUS_BRANCH..HEAD"
+  COMMIT_RANGE="$(git merge-base $CIRRUS_BASE_SHA $CIRRUS_CHANGE_IN_REPO)..$GIT_HEAD"
```

Comment:
Why is this changed again? Now it will include master commits, because master has already been merged in for pull requests (see .cirrus.yml)

`COMMIT_RANGE="${CIRRUS_BASE_SHA}..$GIT_HEAD"` was correct.

Reply:
> ... master has already been merged in for pull requests (see .cirrus.yml)

Right.

For described case "when a changed PR is force pushed into the base branch that has already grown, i.e., the pull is effectively based on the non-top commit of the base branch" I think `COMMIT_RANGE="${CIRRUS_BASE_SHA}..$GIT_HEAD"` is wrong, as `$CIRRUS_BASE_SHA` is not the commit on which the pull is based on.

Reply:
> $CIRRUS_BASE_SHA is not the commit on which the pull is based on.

It is the commit that this pull is de-facto based on, assuming that `git merge A` is approximately doing the same like `git rebase A`.

As I mentioned previously the current version is broken and will include all commits that have been pushed to master since. See:

https://cirrus-ci.com/task/5396234322051072?command=lint#L931

Reply:
In `.cirrus.yml` the base branch is merged into the pull one. That's why I think that [listed](https://cirrus-ci.com/task/5396234322051072?command=lint#L931) commits are correct `COMMIT_RANGE` to lint.

OTOH, we could merge the pull branch into the base one. In that case `$CIRRUS_BASE_SHA` will work as documented.

Reply:
> That's why I think that listed commits are correct COMMIT_RANGE to lint.

It is not the way it worked on travis

Reply:
Agree.

Is it acceptable to s/`git merge A`/`git rebase A`/ ?

Reply:
I've tested `COMMIT_RANGE="${CIRRUS_BASE_SHA}..$GIT_HEAD"`. It works for all cases.
`git` is smart enough to know what to do :)

But for human the value of `COMMIT_RANGE` is a bit cryptic (regarding to pull commits) in cases of a pull based on the non-top commit of the base branch. So going to skip output of it.

Reply:
> Is it acceptable to s/git merge A/git rebase A/ ?

In theory yes, in practise it doesn't work with subtree bumps.